client/checkout/api/index.js

```diff
@@ -51,20 +51,36 @@ export default class WCPayAPI {
 	 * @param {string} paymentMethodType The payment method type.
 	 * @return {Object} The Stripe Object.
 	 */
-	getStripeForUPE( paymentMethodType ) {
+	async getStripeForUPE( paymentMethodType ) {
 		this.options.forceNetworkSavedCards = getUPEConfig(
 			'paymentMethodsConfig'
 		)[ paymentMethodType ].forceNetworkSavedCards;
 		return this.getStripe();
 	}
 
+	async getStripe( forceAccountRequest = false ) {
+		const maxWaitTime = 600 * 1000; // 600 seconds
+		const waitInterval = 100;
+		let currentWaitTime = 0;
+		while ( ! window.Stripe ) {
+			await new Promise( ( resolve ) =>
+				setTimeout( resolve, waitInterval )
+			);
+			currentWaitTime += waitInterval;
+			if ( currentWaitTime > maxWaitTime ) {
+				throw new Error( 'Stripe object not found' );
+			}
+		}
+		return this.__getStripe( forceAccountRequest );
+	}
```

Comment:
While commenting on the `try`/`catch` below, I noticed this implementation: I was wondering if you had considered leveraging [JavaScript Proxies](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) to detect when `Stripe` is added to the `window` object, rather than using a timeout in a loop.